src/Task1/Vote.js

```diff
@@ -1,58 +1,104 @@
 import React, { Component } from 'react'
 import './Vote.css'
 class Vote extends Component {
-    constructor(props) {
-      super(props)
-    
-      this.state =JSON.parse(window.localStorage.getItem('state')) || {
-         languages:[
-            {name:"Javascript",vote:0},
-            {name:"Python",vote:0},
-            {name:"Java",vote:0},
-            {name:"C++",vote:0},
-            {name:"Kotlin",vote:0},
-         ]
-      }
+  constructor(props) {
+    super(props)
+    this.state = {
+      languages: [
+        { name: "Javascript", vote: 0 },
+        { name: "Python", vote: 0 },
+        { name: "Java", vote: 0 },
+        { name: "C++", vote: 0 },
+        { name: "Kotlin", vote: 0 },
+      ]
     }
-    setState(state) {
-        window.localStorage.setItem('state', JSON.stringify(state));
-        super.setState(state);
-      }
-    reset(state){
-       window.localStorage.clear(state);
-       console.log("hello");
-    //    console.log(window.localStorage.getItem('state'));
-      }     
-    vote(i){
-        let newLanguageVote=this.state.languages;
-        console.log(newLanguageVote[i].name);
-        newLanguageVote[i].vote++;
+  }
 
-        console.log("click:",i);
-        console.log(newLanguageVote[i].vote);
-        this.setState({
-            languages:newLanguageVote
-        });
+  //setting localstorage using componentDidMount and ComponentDidUpdate
+  componentDidMount() {
+    if (window.localStorage.getItem('state')) {
+      this.setState({
+        languages: JSON.parse(window.localStorage.getItem('state'))
+      })
     }
-    render() {
-        return (
-        <div>
-            <h1>Vote your languages</h1>
-            <button onClick={this.reset}>Clear All</button>
-            <div className="languages">
-            {
-                this.state.languages.map((item,i)=>
-                <div key={i} className='item'>
-                    <div className='voteCount'>{item.vote}</div>
-                    <div className='lnaguageName'>{item.name}</div>
-                    <button onClick={this.vote.bind(this,i)}>Vote ME</button>
-                </div>
-                )
-            }
-            </div>
+  }
+
+  componentDidUpdate(prevProps, prevState) {
```

Comment:
@Poorvi-3pillar How are prevProps & prevState being used here OR are these 2 helping in any way to set local storage??